x/dex/keeper/place_limit_order.go

```diff
@@ -174,13 +174,13 @@ func (k Keeper) ExecutePlaceLimitOrder(
 			ctx.GasMeter().ConsumeGas(types.ExpiringLimitOrderGas, "Expiring LimitOrder Fee")
 		}
 
-		k.SaveTranche(ctx, placeTranche)
+		k.SaveOrMakeInactiveTranche(ctx, placeTranche)
 
 		totalIn = totalIn.Add(amountLeft)
 		sharesIssued = amountLeft
 	}
 
-	k.SaveTrancheUser(ctx, trancheUser)
+	k.SaveOrRemoveTrancheUser(ctx, trancheUser)
```

Comment:
we're placing a limit order here. can it be the case that we go to the first branch of the `SaveOrRemoveTrancheUser` that does `RemoveLimitOrderTrancheUser` in case `if trancheUser.IsEmpty()`?